### ISSUE TITLE
React Native support

### ISSUE THREAD

**Issue by nicolaslopezj** [nicolaslopezj/meteor-apollo-accounts#2]
This should work in React Native but I think we need little changes in the way we store tokens

https://github.com/nicolaslopezj/meteor-apollo-accounts/blob/master/meteor-apollo-accounts/src/store.js

**Comment by dbrrt** [comment]
I think AsyncStorage might be a good alternative from localStorage for React Native
https://facebook.github.io/react-native/docs/asyncstorage.html

**Comment by nicolaslopezj** [comment]
Nice! Would you like to make a PR?

**Comment by dbrrt** [comment]
Sure !

**Comment by dbrrt** [comment]
Not the best period of the year to code, but I'll try.

**Comment by dbrrt** [comment]
I made a working React Native Authentication example that uses a standalone Apollo Server application (not provided, but can be combined with Janik's) : https://github.com/dbrrt/example_react_native_gql_auth , it's still under development and new features will be added soon; like a view renderable only if user Logged, if not he/she will be only able to mount login view.

**Comment by dbrrt** [comment]
@nicolaslopezj  Concerning the package npm in itself, do you think that something packaged for both React and React-Native is better or there should be two different packages? 
There's at least a solution for something hybrid : https://www.npmjs.com/package/react-native-storage , but I'd rather use a package with more chance to be long term maintained by facebook and its community using directly AsyncStorage.

**Comment by nicolaslopezj** [comment]
I think we should do something like

```js
if (reactNative) {
  AsyncStorage
} else {
  window.localStorage
}

```

**Comment by dbrrt** [comment]
@nicolaslopezj I thought about a solution like that, but there'll be something wrong when you'll import the components, AsyncStorage is React Native specific and need to be explicitly imported.

I don't know if that's syntactically correct, but doing something like that seems really dirty
```javascript
if (reactNative) {
   import {
      ...
      AsyncStorage
    } from 'react-native';
}
```

**Comment by dbrrt** [comment]
@hammadj 
Yes, I tested it last week on React Native and it works well at least with login, it stores the userId in AsyncStorage. Tested on Mac Os X Sierra, React-Native with Apollo-Client (will check the version).
I was waiting for the User Feature in LoginResponse to implement it on my client. I'll have a look on 
Unfortunately, I'm still facing issues with the specific imports (AsyncStorage) that led me to create another folder/library for React-Native.

**Comment by fermat-developer** [comment]
@dbrrt 
I tried the React-Native variant and my app and login seems to work but when I try to query data, the Meteor user is unknown at the server. I used the same query as in my React application:

const query = gql`
query getCurrentUser {
  users {
    _id
    name
  }
}
`
const UserListWithData = graphql(query)(UserList);

Should this work? "getCurrentUser" does not seem to work. I guess "getCurrentUser" uses localStorage and not AsyncStorage?

**Comment by janikvonrotz** [comment]
@fermat-developer `getCurrentUser` is not part of the schema yet, it was merely a proposal to a problem in #23. However, if you require this feature, you can add the query and the resolver to your app yourself.

**Comment by dbrrt** [comment]
@fermat-developer  I'm working on an example for React-Native, but since the library work is still in progress, it's just a Proof of Concept for now.

**Comment by fermat-developer** [comment]
Okay, I got it working. I used Janiks example to run meteor with apollo and meteor accounts, then I npm packed dbrrt's apollo-accounts-react-native fork and added it to a react-native project. At first it did not work because I read the token from AsyncStorage within the applyMiddleware part in ApolloClient. But when I put the applyMiddleware (authorization header) in the AsyncStorage promise it worked. I hope you will create an official npm package for apollo-accounts-react-native.
